samples/api-client/mqtt_example/README.rst

```diff
@@ -0,0 +1,151 @@
+
+
+Authentication
+++++++++++++++
+
+This sample requires you to have authentication setup. Refer to the
+`Authentication Getting Started Guide`_ for instructions on setting up
+credentials for applications.
+
+.. _Authentication Getting Started Guide:
+    https://cloud.google.com/docs/authentication/getting-started
+
+Install Dependencies
+++++++++++++++++++++
+
+#. Clone python-docs-samples and change directory to the sample directory you want to use.
+
+    .. code-block:: bash
+
+        $ git clone https://github.com/GoogleCloudPlatform/python-docs-samples.git
+
+#. Install `pip`_ and `virtualenv`_ if you do not already have them. You may want to refer to the `Python Development Environment Setup Guide`_ for Google Cloud Platform for instructions.
+
+   .. _Python Development Environment Setup Guide:
+       https://cloud.google.com/python/setup
+
+#. Create a virtualenv. Samples are compatible with Python 3.6+.
+
+    .. code-block:: bash
+
+        $ virtualenv env
+        $ source env/bin/activate
+
+#. Install the dependencies needed to run the samples.
+
+    .. code-block:: bash
+
+        $ pip install -r requirements.txt
+
+.. _pip: https://pip.pypa.io/
+.. _virtualenv: https://virtualenv.pypa.io/
+
+Samples
+-------------------------------------------------------------------------------
+
+MQTT example
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+.. image:: https://gstatic.com/cloudssh/images/open-btn.png
+   :target: https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/GoogleCloudPlatform/python-docs-samples&page=editor&open_in_editor=iot/api-client/mqtt_example/cloudiot_mqtt_example.py,iot/api-client/mqtt_example/README.rst
+
+
+
+
+To run this sample:
+
+.. code-block:: bash
+
+    $ python cloudiot_mqtt_example.py
+
+    usage: cloudiot_mqtt_example.py [-h] --algorithm {RS256,ES256}
+                                    [--ca_certs CA_CERTS]
+                                    [--cloud_region CLOUD_REGION] [--data DATA]
+                                    --device_id DEVICE_ID
+                                    [--gateway_id GATEWAY_ID]
+                                    [--jwt_expires_minutes JWT_EXPIRES_MINUTES]
+                                    [--listen_dur LISTEN_DUR]
+                                    [--message_type {event,state}]
+                                    [--mqtt_bridge_hostname MQTT_BRIDGE_HOSTNAME]
+                                    [--mqtt_bridge_port {8883,443}]
+                                    [--num_messages NUM_MESSAGES]
+                                    --private_key_file PRIVATE_KEY_FILE
+                                    [--project_id PROJECT_ID] --registry_id
+                                    REGISTRY_ID
+                                    [--service_account_json SERVICE_ACCOUNT_JSON]
+                                    {device_demo,gateway_send,gateway_listen} ...
```

Comment:
Consider having one argument per line, without a consistent style it can be easily confusing for the users to parse this long information.

Reply:
Some arguments are within square brackets `[]` and some are not. They all seem to be optional arguments and I'm confused if they should be the way they are.

Reply:
This whole thing comes from the readme generation, I cannot change it.